ergo/utils.py

```diff
@@ -74,3 +74,7 @@ def trapz(y, x=None, dx=1.0):
     if x is not None:
         dx = np.diff(x)
     return 0.5 * (dx * (y[..., 1:] + y[..., :-1])).sum(-1)
+
+
+def safe_log(x, constant=1e-15):
```

Comment:
Why did you choose 1e-15? (I would have expected a number closer to the minimum resolution)

```
>>> np.log(1e-37)
DeviceArray(-85.19565, dtype=float32)
>>> np.log(1e-38)
DeviceArray(-inf, dtype=float32)
```